### DISCUSSION TITLE
Full Alert css loading in localhost, not to production site

### DISCUSSION THREAD

**Discussion by raqueeb** [thuliteio/doks#119] [Q&A] [answered]
As per @h-enk instruction, I did the following,

In ./config/_default/config.toml set baseurl = "/"
In ./config/production/config.toml set baseurl = "https://aiwithr.github.io/newsite/"

If I use the relative path assets\scss\common\_fonts.scss for the font,

```
@font-face {
  font-family: "SiyamRupali";
  src: 
    local("Jost"),  
    url('fonts/vendor/jost/SiyamRupali.woff') format('woff');
  font-weight: normal;
  font-style: 400;
  font-display: block;
}
```
Then, it works in the localhost setting, but not in the https://aiwithr.github.io/newsite/. The workaround was to put the font elsewhere and replace url('fonts/vendor/jost/SiyamRupali.woff') format('woff');

with 'https://aiwithr.github.io/css/SiyamRupali.woff', which is constant. But it seems it loads twice as it has local src: local("Jost") too.

![image](https://user-images.githubusercontent.com/32799073/107109101-8196dc80-6867-11eb-9a34-6a000df9b23b.png)

It should change the relative path with the ./config/production/config.toml set baseurl = "https://aiwithr.github.io/newsite/" but does not work for gh-pages. The link is here ... https://aiwithr.github.io/newsite/

The alert works fine in the localhost setting, like this ---
![image](https://user-images.githubusercontent.com/32799073/107109110-95424300-6867-11eb-953b-10103873d391.png)

but for https://aiwithr.github.io/newsite/, it looks like 

![image](https://user-images.githubusercontent.com/32799073/107109215-6d071400-6868-11eb-9eb8-7478be101426.png)

I believe there is some issue with path and,

I tired commenting it out, this part,

```
.alert-light {
  color: #215888;
  background: linear-gradient(-45deg, rgb(212, 245, 255), rgb(234, 250, 255), rgb(234, 250, 255), #d3f6ef);
}

.alert-light .alert-link {
  color: #215888;
}
```
It stays the same.

My generated css file is here -- https://aiwithr.github.io/newsite/main.598ad9e7f0237ae187f65828cc2720fe740790edb400d34b5ff93faab11d6267dafc8d9b1b150dc876a79dd1d44b9deda042a87ab9dfcbc6c5e00ce8a1e76af6.css

Thank you in advance for the assistance.

**Comment by h-enk** [comment, marked as answer]
Re. alert message link: 

If it works OK on localhost, but not in production, it probably has something to do with PurgeCSS.

Add to `./config/postcss.config.js` in the `whitelister` section `'./assets/scss/components/_alerts.scss',`, like so:

```js
const autoprefixer = require('autoprefixer');
const purgecss = require('@fullhuman/postcss-purgecss');
const whitelister = require('purgecss-whitelister');

module.exports = {
  plugins: [
    autoprefixer(),
    purgecss({
      content: [
        './layouts/**/*.html',
        './content/**/*.md',
      ],
      safelist: [
        'lazyloaded',
        ...whitelister([
          './assets/scss/components/_code.scss',
          './assets/scss/components/_search.scss',
          './assets/scss/components/_alerts.scss',
          './assets/scss/common/_dark.scss',
        ]),
      ],
    }),
  ],
}
```
Then, try again.

**Reply by raqueeb** [reply]
Thank you so much! it worked!

**Comment by h-enk** [comment]
@raqueeb, please post 1 issue /question per Doks Discussions item, and please once. That way we keep Doks Discussions clear and better usable. Thanks!

**Reply by raqueeb** [reply]
I'll do that. Thank you.

**Comment by h-enk** [comment]
Re. extra font /double font loading:

Would you mind sharing the original repo (not the result /website repo)?

**Comment by raqueeb** [comment]
You have rightly figured out the problem. Do I need to still upload that?

**Reply by h-enk** [reply]
No, not if your issue(s) are solved 😉
